src/core/index.ts

```diff
@@ -660,10 +665,16 @@ export class ZarrArray {
   }
 
   private encodeChunk(chunk: TypedArray) {
+    if (this.dtype.includes('>')) {
+      // If big endian, flip bytes before applying compression and setting store
+      byteSwap(chunk);
+    }
```

Comment:
Since this will byteswap _in-place_, we should probably copy the chunk if the store is big-endian as to not have side effects on the original input from the end user. Thoughts?

For decoding, I think it's ok to swap bytes in place on the original decoded buffer since the user will never interact with the decoded chunk directly.

Reply:
Yes, let's not touch the original input in case of encoding. For decoding sure we can do it inplace :). Maybe a second function `byteSwapInplace` is a good way to make it explicit?